cmd/api/src/api/v2/agi.go

```diff
@@ -403,27 +404,52 @@ func parseAGMembersFromNodes(nodes graph.NodeSet, selectors model.AssetGroupSele
 		isCustomMember := false
 		// a member is custom if at least one selector exists for that object ID
 		for _, agSelector := range selectors {
-			if agSelector.Selector == node.Properties.Map[common.ObjectID.String()].(string) {
+			if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); !ok {
+				log.Warnf("objectid is missing for node %d", node.ID)
+			} else if agSelector.Selector == objectId {
 				isCustomMember = true
 			}
 		}
 
+		var (
+			memberObjectId string
+			memberName     string
+		)
+
+		if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); ok {
```

Comment:
It is preferred to consistently handle the negative case first and the happy path second